packages/relay/src/lib/eth.ts

```diff
@@ -624,15 +657,31 @@ export class EthImpl implements Eth {
     const latestBlockTolerance = 1;
     this.logger.trace(`${requestIdPrefix} getBalance(account=${account}, blockNumberOrTag=${blockNumberOrTag})`);
 
+    let latestBlock: LatestBlockNumberTimestamp | null | undefined;
     // this check is required, because some tools like Metamask pass for parameter latest block, with a number (ex 0x30ea)
     // tolerance is needed, because there is a small delay between requesting latest block from blockNumber and passing it here
     if (!EthImpl.blockTagIsLatestOrPending(blockNumberOrTag)) {
-      const latestBlock = await this.blockNumber(requestId);
-      const blockDiff = Number(latestBlock) - Number(blockNumberOrTag);
+      const cacheKey = `${constants.CACHE_KEY.ETH_BLOCK_NUMBER}`;
+      const blockNumberCached = this.cache.get(cacheKey);
+      
+      if(blockNumberCached) {
+        this.logger.trace(`${requestIdPrefix} returning cached value ${cacheKey}:${JSON.stringify(blockNumberCached)}`);
+        latestBlock = { blockNumber: blockNumberCached, timeStampTo: '0' };
+
+      } else {
+        latestBlock = await this.blockNumberTimestamp(requestId);
+      }
+      const blockDiff = Number(latestBlock.blockNumber) - Number(blockNumberOrTag);
 
       if (blockDiff <= latestBlockTolerance) {
         blockNumberOrTag = EthImpl.blockLatest;
       }
+
+      // If ever we get the latest block from cache, and blockNumberOrTag is not latest, then we need to get the block timestamp
+      // This should rarely happen.
+      if((blockNumberOrTag !== EthImpl.blockLatest) && (latestBlock.timeStampTo === "0")) {
+        latestBlock = await this.blockNumberTimestamp(requestId);
```

Comment:
do we want to add to the cache here even though this may not happen very often?

Reply:
Well block times are every two seconds and the cached value has a time to live set to one, so I think it makes sense to use the cache most used, hence updated most often, and if that cached value is not the latest then make a call to the mirrornode.